pkg/antctl/raw/check/installation/command.go

```diff
@@ -0,0 +1,405 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package installation
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/cobra"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"antrea.io/antrea/pkg/antctl/raw/check"
+)
+
+func Command() *cobra.Command {
+	o := newOptions()
+	command := &cobra.Command{
+		Use:   "installation",
+		Short: "Runs post installation checks",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return Run(o)
+		},
+	}
+	command.Flags().StringVarP(&o.antreaNamespace, "Namespace", "n", o.antreaNamespace, "Configure Namespace in which Antrea is running")
+	return command
+}
+
+func init() {
+	RegisterTest("Pod-to-Pod Intra-node Connectivity", &PodToPodIntraNodeConnectivityTest{})
+	RegisterTest("Pod-to-Pod Inter-node Connectivity", &PodToPodInterNodeConnectivityTest{})
+	RegisterTest("Pod-to-Internet Connectivity", &PodToInternetConnectivityTest{})
+}
+
+type options struct {
+	antreaNamespace string
+}
+
+func newOptions() *options {
+	return &options{
+		antreaNamespace: "kube-system",
+	}
+}
+
+const (
+	testNamespacePrefix         = "antrea-test"
+	clientDeploymentName        = "test-client"
+	echoSameNodeDeploymentName  = "echo-same-node"
+	echoOtherNodeDeploymentName = "echo-other-node"
+	kindEchoName                = "echo"
+	kindClientName              = "client"
+	agentDaemonSetName          = "antrea-agent"
+	deploymentImage             = "registry.k8s.io/e2e-test-images/agnhost:2.29"
+	podReadyTimeout             = 1 * time.Minute
+)
+
+type Test interface {
+	Run(ctx context.Context, testContext *testContext) error
+	Category() string
+}
+
+var testsRegistry = make(map[string]Test)
+
+func RegisterTest(name string, test Test) {
+	testsRegistry[name] = test
+}
+
+type testContext struct {
+	client           kubernetes.Interface
+	config           *rest.Config
+	clusterName      string
+	antreaNamespace  string
+	clientPods       *corev1.PodList
+	echoSameNodePod  map[string]string
+	echoOtherNodePod map[string]string
+	namespace        string
+}
+
+func Run(o *options) error {
+	client, config, clusterName, err := check.NewClient()
+	if err != nil {
+		return fmt.Errorf("unable to create Kubernetes client: %s", err)
+	}
+	ctx := context.Background()
+	testContext := NewTestContext(client, config, clusterName, o)
+	if err := testContext.setup(ctx); err != nil {
+		return err
+	}
+	for name, test := range testsRegistry {
+		if test.Category() == "inter-node" && !(len(testContext.echoOtherNodePod) > 0) {
```

Comment:
```suggestion
		if test.Category() == "inter-node" && len(testContext.echoOtherNodePod) == 0 {
```
to make it straightforward